File: front/src/api/updateScript.ts
import { Response, checkError, ScriptDetails, ScriptDetailsCreate } from './base';

export const updateScript = async (scriptDetails: ScriptDetailsCreate): Promise<ScriptDetails | null> => {
  try {
    const response = await fetch('/api/v1/script', {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(scriptDetails),
    });
    const data: Response<ScriptDetails> = await response.json();
    return checkError(data);
  } catch (error) {
    console.error(error);
    return null;
  }
};

File: src/api/v1/handlers/update_script.rs
use crate::api::common::models::Response;
use crate::api::common::ApiError;
use crate::script_manager::{self, models::ScriptDetailsCreate};
use actix_web::{web, HttpResponse, ResponseError};

pub async fn update_script(body: web::Bytes) -> HttpResponse {
    let script_details: ScriptDetailsCreate = match serde_json::from_slice(&body) {
        Ok(data) => data,
        Err(e) => {
            return ApiError::new(
                "update_data_file",
                format!("Failed to parse JSON: {}", e),
            )
            .error_response()
        }
    };

    let script_manager = script_manager::get_script_manager();

    match script_manager.update_script(script_details) {
        Ok(updated_script) => HttpResponse::Ok().json(Response::success(updated_script)),
        Err(err) => ApiError::from_debug_error("update_script", err).error_response()
    }
}
